sklearn/utils/validation.py

```diff
@@ -1120,6 +1127,9 @@ def is_sparse(dtype):
             else:
                 array = array.copy(**copy_params)
 
+    if ensure_non_negative:
+        check_non_negative(array, "check_array")
```

Comment:
As stated in another comment, if `input_name` and `estimator` are not the default, I think that we can make a crafted message better than `"check_array"`.